src/Columns/tests/gtest_column_vector.cpp

```diff
@@ -0,0 +1,90 @@
+#include <typeinfo>
+#include <Columns/ColumnsNumber.h>
+#include <Common/randomSeed.h>
+#include <gtest/gtest.h>
+
+
+using namespace DB;
+
+static pcg64 rng(randomSeed());
+static constexpr int error_code = 12345;
+static constexpr size_t TEST_RUNS = 500;
+static constexpr size_t MAX_ROWS = 10000;
+static constexpr size_t filter_ratios[] = {1, 2, 5, 11, 32, 64, 100, 1000};
+static constexpr size_t K = sizeof(filter_ratios) / sizeof(filter_ratios[0]);
+
+template <typename T>
+static MutableColumnPtr createColumn(size_t n)
+{
+    auto column = ColumnVector<T>::create();
+    auto & values = column->getData();
+
+    for (size_t i = 0; i < n; ++i)
+    {
+        values.push_back(i);
+    }
+
+    return column;
+}
+
+bool checkFilter(const PaddedPODArray<UInt8> &flit, const IColumn & src, const IColumn & dst)
+{
+    size_t n = flit.size();
+    size_t dst_size = dst.size();
+    size_t j = 0;   /// index of dest
+    for (size_t i = 0; i < n; ++i)
+    {
+        if (flit[i] != 0)
```

Comment:
Don't you also want to check the opposite condition, i.e. that if `filt[i] == 0` that the original element was not copied into the destination?

Reply:
We already implicitly check the opposite condition.

The `j` is only increase when a match found, if unwanted element copied, it will failed at next check.
We also check the matched size at Line 45 to make sure no illegal elements copied after match finished.